stablehlo/dialect/StablehloOps.cpp

```diff
@@ -1985,6 +1985,28 @@ LogicalResult AllGatherOp::verify() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// AllReduceOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult AllReduceOp::verify() {
+  if (failed(verifyReplicaGroups(*this, /*is_uniform_sized=*/false)))
+    return failure();
+
+  auto operandType = getOperand().getType().cast<TensorType>();
```

Comment:
Need verifications for:
* (C2) `all_gather_dim` $\in$ [0, rank(`operand`)).
* (C3) dim(`operand`, `all_gather_dim`) $\ne$ 0.
* (C4) dim(`result`, `all_gather_dim`) % dim(`operand`, `all_gather_dim`) $=$ 0.

Reply:
Sorry, all these are spurious. removed.